File: Microservices/BillingService/BillingService.Domain/Models/Claims/ClaimsSubmitModel.cs
using Rethink.Services.Common.Enums.Billing;
using System.Collections.Generic;

namespace BillingService.Domain.Models.Claims
{
    public class ClaimsSubmitModel : IdsWithUserInfo
    {
        public bool IsSecondary { get; set; } = false;
        public AdjustmentLevel? AdjustmentLevel { get; set; }
        public List<SecondaryFunderDetailsModel> SecondaryFunderDetails { get; set; }
        public string ImpersonationUserName { get; set; } = string.Empty;
    }
}

File: billing_ui_app/src/app/core/models/billing/claims-submit-model.ts
import { IdsWithUserInfo, UserInfo } from "./get-claim-by-identifier";

export class ClaimsSubmitModel extends IdsWithUserInfo {
    isSecondary : boolean;
    adjustmentLevel : number;
    secondaryFunderDetails : SecondaryFunderDetailsModel[] = [];
    impersonationUserName: string | null;
}

export interface ClearingHouseClaimModel {
  claimId: number;
  clearinghouseId: number;
  isSecondary: boolean;        
  adjustmentLevel?: number;
}

export class SecondaryFunderDetailsModel {
    claimId : number;
    secondaryFunderId : number;
    controlNumber : string;
}
